File: leecode/901-20221021/code.js
var StockSpanner = function() {
    this.stack = [];
};

/** 
 * @param {number} price
 * @return {number}
 */
StockSpanner.prototype.next = function(price) {
    next = [price, 1];
    while(this.stack.length && price >= this.stack[this.stack.length-1][0])
    {
        var tmp = this.stack.pop();
        next[1] += tmp[1];
    }
    this.stack.push(next);
    return next[1];
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * var obj = new StockSpanner()
 * var param_1 = obj.next(price)
 */
const nums = [100, 80, 60, 70, 60, 75, 85];
var stock = new StockSpanner();
for(var i = 0; i < nums.length; i++){
    console.log(stock.next(nums[i]));
}

File: leecode/901-20221021/code.py
from collections import Counter
from typing import List
from cmath import inf

class StockSpanner:

    def __init__(self):
        self.nums = [[inf,0]]
        
    def next(self, price: int) -> int:
        n = len(self.nums)
        i = len(self.nums) - 1
        while price >= self.nums[i][0]:
            i = self.nums[i][1]
        self.nums.append([price, i])
        return n - i



# Your StockSpanner object will be instantiated and called as such:
obj = StockSpanner()
test = [100, 80, 60, 70, 60, 75, 85]
for price in test:
    print(obj.next(price))